common/util/closer.go

```diff
@@ -0,0 +1,22 @@
+package util
+
+// Closer adds the ability to close objects.
+type Closer struct {
```

Comment:
I'd implement an `IsClosed` getter - i think this is the main functionality in this util right?

Reply:
I moved it from the `hare` package without any modifications because I need it as well. Actually, I just need to block until the channel is closed, so `CloseChannel` is enough for me, same in the `hare` package. I've added `IsClosed` in case it's needed in the future.